Switch to uv for RTD build

Zethson opened the pull request:
Fixes https://github.com/scverse/cookiecutter-scverse/issues/399

- Use UV for RTD build which can substantially speed up doc generation & preview

Zethson committed:
Switch to uv for RTD build

Signed-off-by: Lukas Heumos <lukas.heumos@posteo.net>

Zethson committed:
missing empty line

Signed-off-by: Lukas Heumos <lukas.heumos@posteo.net>

Zethson committed:
Stick to 3.12

Signed-off-by: Lukas Heumos <lukas.heumos@posteo.net>

grst commented:
do you by any chance know if there's also a `hatch` plugin for readthedocs? This would tie in well with @flying-sheep and my efforts to standardize builds between local machines and CI. 

We could also execute hatch through uv I suppose.

flying-sheep reviewed (changes requested):
`fail_on_warning` *must* be true by default, and the comment should read “disable this if you are OK with broken links in your docs”. I’m losing my mind here, why does everybody just love to try and change this every single time this file gets touched?

In scanpy I needed to switch it to this before people kept their ill-advised fingers off it.

```yaml
fail_on_warning: true # do not change or you will be fired
```

Besides: does it even do anything if we call `sphinx` manually above? Is there a SPHINX_BUILD_OPTS envvar that gets set by this configuration or something? We should use that or get rid of the whole `sphinx:` block if it does nothing.

and add `-W` to the command line of course.

Zethson commented:
`fail_on_warning must be true by default, and the comment should read “disable this if you are OK with broken links in your docs”. I’m losing my mind here, why does everybody just love to try and change this every single time this file gets touched?`

That's just copy and paste. We'll stick to true :)

Zethson committed:
trueeeeeeeee

Signed-off-by: Lukas Heumos <lukas.heumos@posteo.net>

flying-sheep commented:
phew! this question remains though:

> does [the `sphinx:` config block] even do anything if we call sphinx manually above? Is there a SPHINX_BUILD_OPTS envvar that gets set by this configuration or something? We should use that or get rid of the whole `sphinx:` block if it does nothing.

Zethson committed:
Remove sphinx block

Signed-off-by: Lukas Heumos <lukas.heumos@posteo.net>

Zethson committed:
Refactor

Signed-off-by: Lukas Heumos <lukas.heumos@posteo.net>

Zethson commented:
I don't think that it still does anything. You are correct.

I added `-W` and removed the section.

Reference: https://docs.readthedocs.com/platform/stable/build-customization.html#install-dependencies-with-uv

Zethson commented:
Closing in favor of #402
